### PR TITLE
Change Nomis Timeout to 10 Secs

### PR DESCRIPTION
**Pull request by alexdesi**
Add rescue from error Timeout

### Jira link

https://dsdmoj.atlassian.net/browse/P4-1787

### What?
Reduce timeout to 10s for all NOMIS calls

Ensure we propagate the timeout error message and code (504) to the client for all NOMIS calls

### Why?

To help FrontEnd to provide an appropriate error message

### Have you? (optional)

- [ ] Updated API docs if necessary

### CODE REVIEW

File: app/lib/nomis_client/base.rb [hunk 1]
```diff
@@ -2,17 +2,24 @@
...
+    NOMIS_TIMEOUT = 10 # in seconds
     FIXTURE_DIRECTORY = Rails.root.join 'db/fixtures/nomis'
     MAX_RETRIES = 2
 
     class << self
       def get(path, params = {})
-        benchmark_request(path) { token.get("#{ENV['NOMIS_API_PATH_PREFIX']}#{path}", params) }
+        token.get("#{ENV['NOMIS_API_PATH_PREFIX']}#{path}", params)
+      rescue Faraday::ConnectionFailed, Faraday::TimeoutError => e
+        Rails.logger.info "Nomis Connection Error: #{e.message}"
```
Comment (willfish) [inline app/lib/nomis_client/base.rb:13]:
```suggestion
        Rails.logger.warn "Nomis Connection Error: #{e.message}"
```

File: app/lib/nomis_client/base.rb [hunk 2]
```diff
@@ -2,17 +2,24 @@
...
+        raise e, 'Nomis Connection Error'
       end
 
       def post(path, params = {})
         params = update_json_headers(params)
-        benchmark_request(path) { token.post("#{ENV['NOMIS_API_PATH_PREFIX']}#{path}", params) }
+        token.post("#{ENV['NOMIS_API_PATH_PREFIX']}#{path}", params)
+      rescue Faraday::ConnectionFailed, Faraday::TimeoutError => e
+        Rails.logger.info "Nomis Connection Error: #{e.message}"
```
Comment (willfish) [inline app/lib/nomis_client/base.rb:21]:
```suggestion
        Rails.logger.warn "Nomis Connection Error: #{e.message}"
```

File: app/lib/nomis_client/base.rb [hunk 3]
```diff
@@ -2,17 +2,24 @@
...
+        raise e, 'Nomis Connection Error'
```
Comment (willfish) [inline app/lib/nomis_client/base.rb:22]:
```suggestion
        raise
```

File: app/lib/nomis_client/base.rb [hunk 4]
```diff
@@ -2,17 +2,24 @@
...
     FIXTURE_DIRECTORY = Rails.root.join 'db/fixtures/nomis'
     MAX_RETRIES = 2
 
     class << self
       def get(path, params = {})
-        benchmark_request(path) { token.get("#{ENV['NOMIS_API_PATH_PREFIX']}#{path}", params) }
+        token.get("#{ENV['NOMIS_API_PATH_PREFIX']}#{path}", params)
+      rescue Faraday::ConnectionFailed, Faraday::TimeoutError => e
+        Rails.logger.info "Nomis Connection Error: #{e.message}"
+        raise e, 'Nomis Connection Error'
```
Comment (willfish) [inline app/lib/nomis_client/base.rb:14]:
```suggestion
        raise e
```

File: app/lib/nomis_client/base.rb [hunk 5]
```diff
@@ -2,17 +2,24 @@
 
 module NomisClient
   class Base
+    NOMIS_TIMEOUT = 10 # in seconds
```
Comment (willfish) [inline app/lib/nomis_client/base.rb:5]:
```suggestion
    NOMIS_TIMEOUT = 10.seconds # in seconds
```

File: app/lib/nomis_client/base.rb [hunk 6]
```diff
@@ -2,17 +2,24 @@
...
     FIXTURE_DIRECTORY = Rails.root.join 'db/fixtures/nomis'
     MAX_RETRIES = 2
```
Comment (willfish) [inline app/lib/nomis_client/base.rb:7]: Is this now unused?

File: spec/requests/api/people_controller_index_spec.rb [hunk 7]
```diff
@@ -53,22 +53,34 @@
...
-        expect(response).to have_http_status(:ok)
+          it 'requests data from NOMIS' do
+            expect(response).to have_http_status(:ok)
+          end
+        end
       end
 
-      context 'when the prison_number is downcased' do
-        let(:params) { { filter: { prison_number: prison_number.downcase } } }
+      context 'when Nomis does not reply in less than 10 secs' do
```
Comment (willfish) [inline spec/requests/api/people_controller_index_spec.rb:77]:
```suggestion
      context 'when Nomis times out' do
```

File: app/lib/nomis_client/base.rb (hunk 4 above)
Comment (willfish) [inline app/lib/nomis_client/base.rb:14]:
```suggestion
        raise
```

File: spec/requests/api/people_controller_index_spec.rb (hunk 7 above)
Comment (alexdesi) [reply spec/requests/api/people_controller_index_spec.rb:77]: 👍 that's better

**Comment by alexdesi** [comment]
before merging this, I want to run the E2E test

**Pull request merged**